### PR TITLE
guide.md: added consistency to 04-data-types-and-format section

### PR DESCRIPTION
**Pull request by NPalopoli**
Added Challenge guidelines (taken from the lesson) and language markup to section 04-data-types-and-format for consistency with the other sections.

### CODE REVIEW

File: _extras/guide.md [hunk 1]
```diff
@@ -245,13 +245,20 @@ stack_selection.columns = stack_selection.columns.droplevel()
 
 ### Challenge - Changing Types
 
+* Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Next try converting `weight` to an integer. What goes wrong here? What is Pandas telling you?
+
 Pandas cannot convert types from float to int if the column contains NaN values.
 
 ### Challenge - Counting
+
+* Count the number of missing values per column. Hint: The method .count() gives you the number of non-NA observations per column. Try looking to the .isnull() method.
```
Comment (maxim-belkin) [inline _extras/guide.md]: use backticks around .count() and .isnull() so that they appear as `.count()` and `.isnull()`

File: _extras/guide.md [hunk 2]
```diff
@@ -245,13 +245,20 @@ stack_selection.columns = stack_selection.columns.droplevel()
 
 ### Challenge - Changing Types
 
+* Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Next try converting `weight` to an integer. What goes wrong here? What is Pandas telling you?
```
Comment (maxim-belkin) [inline _extras/guide.md]: Next -> then

Comment (maxim-belkin) [reply _extras/guide.md]: `Then try converting `weight` column to an integer type`

Comment (maxim-belkin) [reply _extras/guide.md]: I'd remove `What goes wrong here?`... and say `What does _pandas_ tell you when you try doing that?` but I'll leave it to @wrightaprilm to decide what is the best wording here

Comment (wrightaprilm) [reply _extras/guide.md]: How about "What error message does Pandas give you? What does this error mean?" Emphasizing specifically that we want them to read and comprehend the output, which in this case is very helpful and informative!

File: _extras/guide.md (hunk 1 above)
Comment (wrightaprilm) [reply _extras/guide.md]:
```suggestion
* Count the number of missing values per column. Hint: The method `.count()` gives you the number of non-NA observations per column. Try looking to the `.isnull()` method.
```

If you just want to accept this change, @NPalopoli, it contains the correct change.

File: _extras/guide.md [hunk 3]
```diff
@@ -245,13 +245,20 @@ stack_selection.columns = stack_selection.columns.droplevel()
...
+* Count the number of missing values per column. Hint: The method .count() gives you the number of non-NA observations per column. Try looking to the .isnull() method.
+
+~~~
 surveys_df.isnull()
+~~~
+{: .language-python}
 
-If the students have trouble generating the output, or anything happens with that, there is a file
-called "sample output" that contains the data file they should generate.
+If the students have trouble generating the output, or anything happens with that, there is a file called "sample output" that contains the data file they should generate.
```
Comment (wrightaprilm) [inline _extras/guide.md]: "there is a folder called `sample_output`, which contains the data file they should generate. It is not included with the data in the setup steps."

Most instructors don't have the learners clone the repo, as Git and GitHub are not part of the Data Carpentry core. So learners typically don't have this file, but the instructor could provide it.

Comment (NPalopoli) [reply _extras/guide.md]: Nice call April. Perhaps this is a good wording?

```suggestion
If the students have trouble generating the output, or anything happens with that, there is a file called "sample output" that contains the data file they should generate. It is not included with the data in the setup steps, so remember to have it at hand and provide it if needed.
```

File: _extras/guide.md (hunk 2 above)
Comment (NPalopoli) [reply _extras/guide.md]: Thanks @maxim-belkin  and @wrightaprilm. Is this a good wording to commit?

```suggestion
* Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Then, try converting the contents of the `weight` column to an integer type. What error message does Pandas give you? What does this error mean?
```

File: _extras/guide.md (hunk 3 above)
Comment (maxim-belkin) [reply _extras/guide.md]: @NPalopoli, note "folder" vs "file".
Another note: "there is" does not answer the question of "where". I suggest saying something like `This repository has a folder called...`

**Comment by maxim-belkin** [comment]
Please change all `Try doing something` to `Do something`, for example: `Try converting` to `Convert`.

File: _extras/guide.md (hunk 2 above)
Comment (wrightaprilm) [reply _extras/guide.md]: I like this wording!

**Comment by wrightaprilm** [comment]
I actually like the `try` language, particularly for things where we are asking for an action that we know will produce an error.

**Comment by maxim-belkin** [comment]
Ha! Agreed! Was not realizing we're talking about stuff that is going to fail. Please leave those `Try`.

File: _extras/guide.md (hunk 3 above)
Comment (NPalopoli) [reply _extras/guide.md]: Thank you @maxim-belkin once and again! Should be correct now.

**Comment by wrightaprilm** [comment]
These changes satisfy my concerns. I'm happy if you're happy, @maxim-belkin.

File: _extras/guide.md [hunk 4]
```diff
@@ -245,13 +245,20 @@ stack_selection.columns = stack_selection.columns.droplevel()
 
 ### Challenge - Changing Types
 
+* Try converting the column `plot_id` to floats using `surveys_df.plot_id.astype("float")`. Then, try converting the contents of the `weight` column to an integer type. What error message does Pandas give you? What does this error mean?
```
[the remark this answers is not in the document]
Comment (NPalopoli) [reply _extras/guide.md]: Already committed those changes. Thank you for pointing out @maxim-belkin!

**Comment by wrightaprilm** [comment]
Great, thanks everyone!

**Pull request merged**